modules/notification-workflowoperation/src/main/java/org/opencastproject/workflow/handler/notification/EmailWorkflowOperationHandler.java

```diff
@@ -94,28 +100,33 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
     MediaPackage srcPackage = workflowInstance.getMediaPackage();
 
     // "To", "CC", "BCC", subject, body can be Freemarker templates
-    String to = processDestination(workflowInstance, operation, TO_PROPERTY);
-    String cc = processDestination(workflowInstance, operation, CC_PROPERTY);
-    String bcc = processDestination(workflowInstance, operation, BCC_PROPERTY);
-    String subject = applyTemplateIfNecessary(workflowInstance, operation, SUBJECT_PROPERTY);
-    String bodyText = null;
+    String[] to = processDestination(workflowInstance, operation, TO_PROPERTY);
+    String[] cc = processDestination(workflowInstance, operation, CC_PROPERTY);
+    String[] bcc = processDestination(workflowInstance, operation, BCC_PROPERTY);
+
+    if (to.length + cc.length + bcc.length == 0) {
+      logger.info("No recipients. Skipping operation.");
+      return createResult(srcPackage, Action.SKIP);
+    }
+
+    String subject = applyTemplateIfNecessary(workflowInstance, operation, SUBJECT_PROPERTY, ", ");
```

Comment:
Why ", " and not let the service use its default DEFAULT_DELIMITER_FOR_MULTIPLE? Just curious. Also, I think this should be a constant as it's used in a few places.

Reply:
My idea was that if you have two subjects, the resulting text would just look more natural like this:

- this is a subject a, this is subject b

rather than

- this is a subject a,this is subject b

But this should seldom happen and honestly, I don't really care either way. What would you prefer?

- Using `, ` (as a local const)
- Using the current default `,`
- Or even making `, ` the new default

Pick your favorite ;-)

Reply:
I don't care which one either, but I it should be a constant (local or not). So how about option 3 to simplify?

Reply:
@rute-santos, done. A new default is set and the inline strings have been removed. Can you take another shot at the review?